### PR TITLE
Merge changes from the private repo.

### PR DESCRIPTION
**Pull request by taegyunkim**
No single occurrence of the word 'tofino' except in this description. 

Created in following steps
1. In local repository, add private repo as remote.
2. git diff master remotes/private/master > ../private.diff
3. git apply ../private.diff
4. Manually inspect each file and modify as needed. 
5. Run tests.

### CODE REVIEW

File: chipc/alu.g4 [hunk 1]
```diff
@@ -23,6 +29,9 @@ LESS_OR_EQUAL    : '<=';
 NOT_EQUAL        : '!=';
 OR               : '||';
 AND              : '&&';
+NOT              : '!';
+QUESTION         : '?';
```
Comment (XiangyuG) [inline chipc/alu.g4:33]: What is the function of QUESTION?

Comment (taegyunkim) [reply chipc/alu.g4:33]: Looks like it's not used anywere, probably safe to remove.

File: example_specs/simplest.sk [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// Spec for Sketch
```
Comment (XiangyuG) [inline example_specs/simplest.sk:1]: Why do we need to add simplest.sk and simplest2.sk? For Tofino experiment? Maybe we could add them into test in private repo.

File: chipc/alu.g4 (hunk 1 above)
Comment (XiangyuG) [reply chipc/alu.g4:33]: Is it because of the this statement "expr '?' expr ':' expr #Ternary"?

File: example_specs/simplest.sk (hunk 2 above)
Comment (taegyunkim) [reply example_specs/simplest.sk:1]: Yes, it was added for tofino experiment. I plan to remove all non-tofino related code and files from tofino repo. chipmunk should be able to compile this, so I think there is no harm in adding this here.

File: chipc/alu.g4 (hunk 1 above)
Comment (XiangyuG) [reply chipc/alu.g4:33]: Line 121 in chipc/alu.g4

Comment (taegyunkim) [reply chipc/alu.g4:33]: Ternary gets separate rule for parsing. We can deal with these minor details in a separate PR.

**Review by XiangyuG — APPROVED** [review]

File: example_specs/simplest.sk (hunk 2 above)
Comment (XiangyuG) [reply example_specs/simplest.sk:1]: Sure~ I see.

**Review by XiangyuG — APPROVED** [review]

**Comment by taegyunkim** [comment]
Normally, I would suggest rebasing each branch on top of the master, if the master has changed.

**Pull request merged**